state/protocol/badger/mutator.go

```diff
@@ -981,6 +996,73 @@ func (m *FollowerState) epochStatus(block *flow.Header, epochFallbackTriggered b
 
 }
 
+// versionBeaconOnBlockFinalized extracts the VersionBeacons from the parent block's
+// Seals and returns it.
```

Comment:
The `header` is the finalized block, why finding the events from the parent of the finalized block, instead of the finalized block itself?